src/pages/ConvertVrt/Convert/index.tsx

```diff
@@ -1,141 +1,218 @@
+              {xvsAmountWei &&
+                convertWeiToCoins({
+                  valueWei: xvsAmountWei,
+                  tokenId: XVS_ID,
+                  returnInReadableFormat: true,
+                })}
+            </Typography>
+          </div>
+        ),
+      });
+    } catch (err) {
+      toast.error({ title: (err as Error).message });
+    }
+  };
+  const userVrtBalance =
+    userVrtBalanceWei &&
+    convertWeiToCoins({ valueWei: userVrtBalanceWei, tokenId: VRT_ID }).toFixed();
   return (
     <div css={styles.root}>
       {walletConnected ? (
         <section css={styles.title}>
-          <Typography variant="h3">{readableXvsAvailable}</Typography>
+          <Typography variant="h3">{readableXvsAvailable || PLACEHOLDER_KEY}</Typography>
           <Typography variant="small2">{t('convertVrt.xvsAVailable')}</Typography>
         </section>
       ) : (
         <div css={styles.smallSpacer} />
       )}
       <ConnectWallet message={t('convertVrt.connectWalletToConvertVrtToXvs')}>
-        <AmountForm onSubmit={noop} maxAmount={vrtLimitWei.toFixed()} css={styles.form}>
-          {({ values }) => {
-            const xvsValue = values.amount
-              ? new BigNumber(values.amount).times(xvsToVrtRate).dp(VRT_DECIMAL).toFixed()
-              : '';
-            return (
-              <>
-                <div css={styles.inputSection}>
-                  <Typography variant="small2" css={styles.inputLabel}>
-                    {t('convertVrt.convertVrt')}
-                  </Typography>
-                  <FormikTokenTextField
-                    tokenId={VRT_ID}
-                    name="amount"
-                    css={styles.input}
-                    description={
-                      <Trans
-                        i18nKey="convertVrt.balance"
-                        components={{
-                          White: <span css={styles.whiteLabel} />,
-                        }}
-                        values={{ amount: format(new BigNumber(vrtLimitWei)) }}
-                      />
+        <EnableToken
+          title={t('convertVrt.enableVrt')}
+          assetId={VRT_ID}
+          isEnabled={userVrtEnabled}
+          tokenInfo={[
+            {
+              iconName: 'xvs',
+              label: 'VRT Conversion Ratio',
+              children: xvsToVrtConversionRatio?.toFixed(6),
+            },
+            { iconName: 'vrt', label: 'Current VRT Balance', children: readableUserVrtBalance },
+          ]}
+          vtokenAddress={vrtConverterProxyAddress}
+        >
+          <AmountForm onSubmit={onSubmit} maxAmount={userVrtBalance} css={styles.form}>
+            {({ values }) => {
+              const xvsValue =
+                values.amount && xvsToVrtConversionRatio
+                  ? new BigNumber(values.amount)
+                      .times(xvsToVrtConversionRatio)
+                      .dp(VRT_DECIMAL)
+                      .toFixed()
+                  : '';
```

Comment:
How about making this into a function wrapped with `React.useCallback`, outside the render block, that can then be called to get the converted XVS amount?

Reply:
I made one but it got dropped here during a rebase it looks like.